gcd/src/main/java/org/spine3/server/storage/datastore/dsnative/NamespaceSupplier.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2017, TeamDev Ltd. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package org.spine3.server.storage.datastore.dsnative;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.spine3.server.storage.datastore.DatastoreStorageFactory;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * A factory for the {@linkplain Namespace namespaces}, based on the current multitenancy configuration and
+ * {@linkplain org.spine3.users.TenantId tenant ID}.
+ *
+ * @author Dmytro Dashenkov
+ */
+public abstract class NamespaceSupplier {
```

Comment:
Since we've made the `DatastoreWrapper` `public` and overridable, we need to disclosure `NamespaceSupplier` as well, since `DatastoreWrapper` receives it in the constructor.
Also, one can think of extending this class as about a feature, allowing additional configurations, groupings, etc. for the records in the DB.